pkg/remote/kyma_synchronization_context.go

```diff
@@ -172,10 +198,11 @@ func (c *KymaSynchronizationContext) CreateOrFetchRemoteKyma(
 
 	err := c.RuntimeClient.Get(ctx, client.ObjectKeyFromObject(remoteKyma), remoteKyma)
 
-	if meta.IsNoMatchError(err) {
-		recorder.Event(kyma, "Normal", err.Error(), "CRDs are missing in SKR and will be installed")
-
-		if err := c.CreateOrUpdateCRD(ctx, v1beta2.KymaKind.Plural()); err != nil {
+	if meta.IsNoMatchError(err) || err == nil {
+		if meta.IsNoMatchError(err) {
+			recorder.Event(kyma, "Normal", err.Error(), "CRDs are missing in SKR and will be installed")
+		}
+		if err = c.CreateOrUpdateCRD(ctx, v1beta2.KymaKind.Plural(), kyma); err != nil {
```

Comment:
can you use this `CreateOrUpdateCRD` to replace with `CreateModuleTemplateCRDInRuntime`, they basically do same thing